src/Catalog/Persistence/AzureStorage.cs

```diff
@@ -16,6 +16,7 @@ namespace NuGet.Services.Metadata.Catalog.Persistence
     public class AzureStorage : Storage
     {
         private readonly CloudBlobDirectory _directory;
+        private readonly int MaxCountOfBlobSnapshots = 30;
```

Comment:
In order to eliminate the perf impact, the design changed to have only one snapshot per blob.